tests/integration/api/endpoints/test_logging_endpoint.py

```diff
@@ -1,32 +1,74 @@
-import pytest
-import os
+import uuid
+import time
+import random
 import asyncio
+import pathlib
+from concurrent.futures import ThreadPoolExecutor
+
+import pytest
 
-from httpx import AsyncClient
-from jinad.config import log_config
+from jinad.excepts import HTTPException
+from jinad.config import log_config, fastapi_config
 
-cur_dir = os.path.dirname(os.path.abspath(__file__))
 LOG_MESSAGE = 'log message'
 
 
-async def feed_path_logs(path):
-    for i in range(0, 100):
-        os.makedirs(path, exist_ok=True)
-        file = os.path.join(path, 'log.log')
-        with open(file, 'a') as fp:
-            fp.write(LOG_MESSAGE)
+def feed_path_logs(filepath, total_lines, sleep):
+    # method to write logs to a file in random interval
+    # this runs in a separate thread
+    pathlib.Path(filepath).parent.absolute().mkdir(parents=True, exist_ok=True)
+    with open(filepath, 'a') as fp:
+        for _ in range(total_lines):
+            fp.writelines(LOG_MESSAGE + '\n')
             fp.flush()
-            await asyncio.sleep(0.1)
+            time.sleep(sleep)
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('total_lines, num_lines_to_fetch, sleep', [
+    (10, 5, 0.1), (10, 5, random.random()),
+    (20, 20, 0.1), (20, 20, random.random()),
+    (100, 150, 0.1), (100, 150, random.random())
+])
+async def test_logging_endpoint_success(fastapi_client, total_lines, num_lines_to_fetch, sleep):
```

Comment:
this should read a file that is continuously being added right?

Reply:
yah. we are doing the same in `feed_path_logs()` in a different thread.